util/kvcache/simple_lru.go

```diff
@@ -87,7 +87,8 @@ func (l *SimpleLRUCache) Get(key Key) (value Value, ok bool) {
 }
 
 // Put puts the (key, value) pair into the LRU Cache.
-func (l *SimpleLRUCache) Put(key Key, value Value) {
+// Evicted key value pair will be processed by func optKvFunc
+func (l *SimpleLRUCache) Put(key Key, value Value, optKvFunc ...func(k Key, v Value)) {
```

Comment:
I don't like this signature. It assumes that the evict is always happened synchronously with PUT, making LRU cache hard to be optimized later. And it is also very rare that there are multiple "on evict" handlers.

Please design from scratch (instead of trying to make workarounds based on current situation), consider the best design solutions, and then compare whether each solution is feasible. (and if no solution is perfect then compare pros and cons).